frame/staking/src/pallet/mod.rs

```diff
@@ -49,6 +49,7 @@ use crate::{
 };
 
 const STAKING_ID: LockIdentifier = *b"staking ";
+pub(crate) const SPECULATIVE_NUM_SPANS: u32 = 10_000;
```

Comment:
This might be a bit too much though :D When you run the benchmarks, the weight value you get is coming from this value. I will think about what could be a good "guess" here.